docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplate.java

```diff
@@ -48,43 +48,53 @@
     private static final Logger LOGGER = Logger.getLogger(DockerTemplate.class.getName());
 
 
-    public final String labelString;
+    private final String labelString;
 
     // SSH settings
     /**
      * The id of the credentials to use.
      */
+    @Deprecated
     public final String credentialsId;
 
     /**
      * Minutes before terminating an idle slave
+     * @deprecated  migrated to retention strategy?
      */
-    public final String idleTerminationMinutes;
+    @Deprecated
+    private String idleTerminationMinutes;
 
     /**
      * Minutes before SSHLauncher times out on launch
      */
-    public final String sshLaunchTimeoutMinutes;
+    @Deprecated
+    private String sshLaunchTimeoutMinutes;
 
     /**
      * Field jvmOptions.
      */
-    public final String jvmOptions;
+    @Deprecated
```

Comment:
probably want to make all these `@Deprecated` fields as `transient` so that they do not get saved out on resave